.gitignore

```diff
@@ -13,3 +13,6 @@ examples/warToJson/files/output/other/*.json
 coverage/
 .nyc_output/
 test/.output
+**/*d.ts
```

Comment:
Should also ignore any of the `.js` files that are generated.
E.g. `lib/translators/SoundsTranslator.js` should be ignored, since the source code is now `SoundsTranslator.ts`, and the JavaScript will be built by the developer in their local environment. Same goes for the other translators, plus `HexBuffer.js`, `W3Buffer.js` since those are backed by TS files too.

Reply:
Ah there it is mhm but then npm wont have js code? We still want to provide the js code for npm, no?

Reply:
Won't be necessary to provide js code on npm -- developers will either build WC3MapTranslator via `tsc` if they're working on the project directly, or they will import the translators into some other project, and the build process on their project will take care of handling WC3MapTranslator for them.

What we can do though, if you're thinking of having the JS files ready to go when the developer does an `npm install` for WC3MapTranslator, is to have a `postinstall` script in `package.json` to execute a TypeScript build. This way when they download this project the JS files will be ready for them out-of-the-box, without us having to handle both a TS and JS version in the source code, plus possibly forgetting to build before pushing updates to npm.

Reply:
Won't be necessary to provide js code on npm -- developers will either build WC3MapTranslator via `tsc` if they want JS code to work with, or they will import the translators into some other project, and the build process on their project will take care of handling WC3MapTranslator for them.

What we can do though, if you're thinking of having the JS files ready to go when the developer does an `npm install` for WC3MapTranslator, is to have a `postinstall` script in `package.json` to execute a TypeScript build. This way when they download this project the JS files will be ready for them out-of-the-box, without us having to handle both a TS and JS version in the source code, plus possibly forgetting to build before pushing updates to npm.